### PR TITLE
feat: allow `ignoredHostnames` to exlude all sub-domains to a root domain

### PR DESCRIPTION
**Pull request by maximepvrt**
Currently, `ignoredHostnames` option not allow exlude a root domain.

Example:
domain: admin.benevolt.dev
```
ignoredHostnames: ['benevolt.dev']
```

### CODE REVIEW

**Comment by Barbapapazes** [comment]
Hey,

Could you give more informations about your use case because I'm not sure to understand why you would need to ignore every sub domain? Why not just adding `admin.benevolt.dev` to the array?

**Comment by maximepvrt** [comment]
I have more than 30 subdomains on my staging app with `benevolt.dev` and the list of these subdomains is managed in the database.

In addition, many use proxies locally with subdomains of `.localhost` or `.test`, this addition allows automatic ignoring.

**Comment by maximepvrt** [comment]
@Barbapapazes ;-)

File: src/plausible.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ export function createPlausibleTracker(initOptions?: Partial<PlausibleOptions>)
     hashMode: false,
     domain: location.hostname,
     apiHost: 'https://plausible.io',
-    ignoredHostnames: ['localhost'],
+    ignoredHostnames: ['localhost', 'test'],
```
Comment (Barbapapazes) [inline src/plausible.ts:25]:
```suggestion
    ignoredHostnames: ['localhost'],
```

**Comment by Barbapapazes** [comment]
> @Barbapapazes ;-)

Hey @maximepvrt! Sorry for the late response, I'm on it. Expect something in less than an hour.

**Comment by Barbapapazes** [comment]
Hey,

I've refactor in order to avoid breaking change.

You can use the new option `ignoreSubDomains`.

**Pull request closed without merging**